### PR TITLE
16 add encryption for cnc files

### PR DESCRIPTION
**Pull request by georgulbrich-isg**
Pull encryption logic into ISG extension.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -26,9 +26,16 @@
 				"--extensionTestsPath=${workspaceFolder}/out/test/suite/index"
 			],
 			"outFiles": [
-				"${workspaceFolder}/out/test/**/*.js"
+				"${workspaceFolder}/out/**/*.js"
 			],
-			"preLaunchTask": "${defaultBuildTask}"
+			"preLaunchTask": "${defaultBuildTask}",
+			"autoAttachChildProcesses": true,
+
```
Comment (georgulbrich-isg) [inline .vscode/launch.json:33]: Leerzeichen

File: media/vscode.css [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/*
```
Comment (georgulbrich-isg) [inline media/vscode.css:1]: Remove file.

File: media/main.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+body {
```
Comment (georgulbrich-isg) [inline media/main.css:1]: Remove file.

File: package-lock.json [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
     "lockfileVersion": 2,
     "requires": true,
     "packages": {
         "": {
             "name": "vscode-isg-cnc",
-            "version": "0.2.7",
+            "version": "0.2.8",
             "license": "SEE LICENSE IN LICENSE",
             "devDependencies": {
+                "@types/ffi-napi": "^4.0.5",
```
Comment (georgulbrich-isg) [inline package-lock.json:12]: Remove ffi-napi.

File: package-lock.json [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
                 "@types/mocha": "^9.0.0",
-                "@types/node": "14.x",
+                "@types/node": "^14.18.18",
+                "@types/open": "^6.2.1",
                 "@types/vscode": "^1.59.0",
                 "@typescript-eslint/eslint-plugin": "^4.31.1",
                 "@typescript-eslint/parser": "^4.31.1",
                 "@vscode/test-electron": "^1.6.2",
                 "eslint": "^7.32.0",
+                "ffi-napi": "^4.0.3",
```
Comment (georgulbrich-isg) [inline package-lock.json:22]: Remove ffi-napi.

File: package-lock.json [hunk 6]
```diff
@@ -222,6 +271,17 @@
                 "node": ">= 6"
             }
         },
+        "node_modules/@types/ffi-napi": {
```
Comment (georgulbrich-isg) [inline package-lock.json:274]: Remove ffi-napi.

File: src/util/encryption/blowfish.ts [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+    blowfish = new Blowfish(key);
+  } else {
+    blowfish = new Blowfish(new TextEncoder().encode(key));
+  }
+
+  const encodedBuffer: Buffer = fs.readFileSync(inputPath);
+  let encodedArray: Uint8Array = new Uint8Array(encodedBuffer);
+
+  let decodedArray: Uint8Array = blowfish.decode(encodedArray);
+
```
Comment (georgulbrich-isg) [inline src/util/encryption/blowfish.ts:427]: Remove blank line.

File: src/util/encryption/encryption.ts [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+        const outputPath: string = path.join(path.dirname(inputPath), outputName);
+        blowfish.encryptFileToFile(inputPath, outputPath, key);
+        vscode.window.showInformationMessage(inputName + " was encrypted into " + outputName);
+    } else {
+        vscode.window.showInformationMessage("Encryption canceled");
+    }
+
+}
+
+
```
Comment (georgulbrich-isg) [inline src/util/encryption/encryption.ts:78]: blanks

File: src/util/encryption/encryption.ts [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+            endSelection = true;
+        }
+
+
+    } while (!endSelection);
+
+
+    return finalOutputName;
+}
+/**
```
Comment (georgulbrich-isg) [inline src/util/encryption/encryption.ts:222]: blanks

**Review by georgulbrich-isg — COMMENTED** [review]
Also: update changelog.

**Comment by jurekseverin-isg** [comment]
The latest commit should have done it.

**Pull request merged**